packages/components/src/button/index.js

```diff
@@ -28,7 +28,7 @@ export function Button( props, ref ) {
 
 	const classes = classnames( 'components-button', className, {
 		'is-button': isDefault || isPrimary || isLarge || isSmall,
-		'is-default': isDefault || isLarge || isSmall,
+		'is-default': ! isPrimary && ( isDefault || isLarge || isSmall ),
```

Comment:
Would the condition `isDefault || ! isPrimary && ( isLarge || isSmall )`also work? I guess when someone passes `isDefault` there is an expectation the is-default class is added for sure. Otherwise, maybe we can add a small note on  https://github.com/WordPress/gutenberg//blob/dd9bf47ff07680d672efc5b31abac268e2cb69c0/packages/components/src/button/README.md#L128 saying the prop is ignored if isPrimary prop is also passed.

Reply:
Hm, well, the root of the problem is that nobody should ever apply _both_ `isPrimary` and `isDefault`, so your expectation may be valid in considering `isDefault` alone, but not in combination with `isPrimary`.

That being said, your suggestion still seems like a slight improvement in sensibly mapping `isDefault` and `isPrimary` to their respective classes, irrespective the styling conflict which would likely ensue from this choice.

Reply:
Updated in cc9a75540.